actions.go

```diff
@@ -419,31 +419,34 @@ type GetenvFunc func(key string) string
 
 // Context of current workflow
 type Context struct {
-	Event      map[string]interface{}
-	EventName  string
-	SHA        string
-	Ref        string
-	Workflow   string
-	Action     string
-	Actor      string
-	Job        string
-	RunNumber  int
-	RunID      int
-	APIURL     string
-	ServerURL  string
-	GraphqlURL string
+	Event            map[string]interface{}
+	EventPayloadPath string `env:"GITHUB_EVENT_PATH,required"`
+	EventName        string
+	SHA              string `env:"GITHUB_SHA"`
+	Ref              string `env:"GITHUB_REF"`
+	Workflow         string `env:"GITHUB_WORKFLOW"`
+	Action           string `env:"GITHUB_ACTION"`
+	Actor            string `env:"GITHUB_ACTOR"`
+	Job              string `env:"GITHUB_JOB"`
+	RunNumber        int    `env:"GITHUB_RUN_NUMBER"`
+	RunID            int    `env:"GITHUB_RUN_ID"`
+	APIURL           string `env:"GITHUB_API_URL,default=https://api.github.com"`
+	ServerURL        string `env:"GITHUB_SERVER_URL,default=https://github.com"`
+	GraphqlURL       string `env:"GITHUB_GRAPHQL_URL,default=https://api.github.com/graphql"`
 }
 
 // Context returns the context of current action with the payload object that triggered the workflow
-func (c *Action) Context() (*Context, error) {
-	var rawEvent map[string]interface{}
-
-	eventPath := c.Getenv("GITHUB_EVENT_PATH")
-	if eventPath == "" {
-		return nil, fmt.Errorf("missing GITHUB_EVENT_PATH in environment")
+func (c *Action) Context(ctx context.Context) (*Context, error) {
+	var (
```

Comment:
Do we still need this?

Reply:
not sure if I get it, the variables? Yes

Reply:
don't worry I can clean that for you